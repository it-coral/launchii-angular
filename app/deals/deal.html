<div ui-view="page_body">
    <div class="row mb-2">
        <div class="col-md-12">
            <a ui-sref="dashboard.deal.add">
                <button type="button" class="btn btn-primary"><i class="fa fa-plus"></i> New Deal</button>
            </a>
        </div>
    </div>
    <div class="row">
        <div class="col-md-12">
            <div class="portlet box green">
                <div class="portlet-title">
                    <div class="caption">
                        <i class="fa fa-comments"></i>List
                    </div>
                    <div class="col-sm-3 mt-1 pull-right">
                        <form ng-submit="vm.startSearch()" novalidate="novalidate">
                            <div class="form-group mb-0">
                                <div class="input-group search-input">
                                    <input type="text" class="form-control" ng-model="vm.searchTerm" placeholder="Search deals..." disable-valid-styling="true" />
                                    <span class="input-group-addon" ng-click="vm.startSearch()">
                                    <i class="icon-magnifier font-white"></i>
                                    </span>
                                    <span class="input-group-addon" ng-click="vm.clearSearch()" ng-if="vm.searchTerm">
                                    <i class="fa fa-close font-white"></i>
                                    </span>
                                </div>
                            </div>
                        </form>
                    </div>
                    <div class="col-sm-2 mt-1 pull-right">
                        <select class="form-control col-xs-12" ng-model="vm.filterDealStatus">
                            <option value="">Status...</option>
                            <option value="draft">Draft</option>
                            <option value="published">Published</option>
                            <option value="hidden">Hidden</option>
                            <option value="archived">Archived</option>
                            <option value="pending">Pending</option>
                            <option value="approved">Approved</option>
                            <option value="rejected">Rejected</option>
                        </select>
                    </div>
                    <div class="col-sm-2 mt-1 pull-right">
                        <select class="form-control col-xs-12" ng-model="vm.filterDealType">
                            <option value="">Deal type...</option>
                            <option value="standard">Standard</option>
                            <option value="upsell">Upsell</option>
                        </select>
                    </div>
                </div>
                <div class="portlet-body">
                    <div class="alert {{vm.response.success}}" ng-if="vm.response.msg">
                        <strong>{{vm.response.alert}}</strong> {{vm.response.msg}}
                    </div>
                    <div class="row table-scrollable">
                        <table class="table table-striped table-hover">
                            <thead>
                                <tr>
                                    <th> Name </th>
                                    <th> Status </th>
                                    <th> Brand </th>
                                    <th> Link </th>
                                    <th> Price </th>
                                    <th> Action </th>
                                </tr>
                            </thead>
                            <tbody>
                                <tr ng-repeat="deal in vm.deals" ng-show="!vm.isLoading && vm.deals.length != 0">
                                    <td ng-bind="deal.name"></td>
                                    <td ng-bind="deal.status" style="text-transform: capitalize;"></td>
                                    <td ng-bind="deal.brand.name"></td>
                                    <td>
                                        <a href="{{deal.link}}" ng-bind="deal.link" target="_blank"></a>
                                    </td>
                                    <td ng-bind="deal.price | toCurrencyFormat"></td>
                                    <td>
                                        <a ui-sref="dashboard.deal.view({id:deal.uid})" class="btn btn-outline btn-circle btn-sm blue">
                                            <i class="fa fa-share"></i> View
                                        </a>
                                        <a ng-href="{{vm.customerHost}}/deals/{{deal.token}}/preview" target="_blank" class="btn btn-outline btn-circle btn-sm green">
                                            <i class="fa fa-eye"></i> Preview
                                        </a>
                                        <a ui-sref="dashboard.deal.edit({id:deal.uid})" class="btn btn-outline btn-circle btn-sm purple">
                                            <i class="fa fa-edit"></i> Edit
                                        </a>
                                        <a href="javascript:;" ng-click="vm.deleteDeal($event.currentTarget, deal)" data-spinner-color="#e7505a" class="btn btn-outline btn-circle btn-sm red">
                                            <i class="fa fa-trash-o"></i> Delete
                                        </a>
                                    </td>
                                </tr>
                                <tr ng-show="!vm.isLoading && vm.deals.length == 0">
                                    <td colspan="6">
                                        <div class="text-center">
                                            <h3>No deals</h3>
                                            <a ui-sref="dashboard.deal.add">
                                                <button type="button" class="btn btn-primary"><i class="fa fa-plus"></i> New Deal</button>
                                            </a>
                                        </div>
                                    </td>
                                </tr>
                                <tr ng-show="vm.isLoading">
                                    <td colspan="6">
                                        <ng-include src="'/app/common/loading.html'"></ng-include>
                                    </td>
                                </tr>
                            </tbody>
                        </table>
                    </div>
                </div>
            </div>
        </div>
    </div>
</div>
