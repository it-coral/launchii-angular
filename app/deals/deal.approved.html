<div ui-view="page_body">
    <div class="row">
        <div class="col-md-12">
            <div class="portlet box green">
                <div class="portlet-title">
                    <div class="caption">
                        <i class="fa fa-comments"></i>List
                    </div>
                </div>
                <div class="portlet-body">
                    <div class="alert {{vm.response.success}}" ng-if="vm.response.msg">
                        <strong>{{vm.response.alert}}</strong> {{vm.response.msg}}
                    </div>
                    <div class="table-scrollable" ng-if="!vm.isDealEmpty">
                        <table class="table table-striped table-hover">
                            <thead>
                                <tr>
                                    <th> Name </th>
                                    <th> Status </th>
                                    <th> Link </th>
                                    <th> Price </th>
                                    <th> Action </th>
                                </tr>
                            </thead>
                            <tbody>
                                <tr ng-repeat="deal in vm.deals" ng-show="vm.isLoading == false">
                                    <td ng-bind="deal.name"></td>
                                    <td ng-bind="deal.status" style="text-transform: capitalize;"></td>
                                    <td>
                                        <a href="{{deal.link}}" ng-bind="deal.link" target="_blank"></a>
                                    </td>
                                    <td ng-bind="deal.price | toCurrencyFormat"></td>
                                    <td>
                                        <a href="javascript:;" ng-click="vm.publishDeal($event.currentTarget, deal)" data-spinner-color="#e7505a" class="btn btn-outline btn-circle btn-sm green">
                                            Publish
                                        </a>
                                    </td>
                                </tr>
                                <tr ng-show="!vm.isLoading && vm.deals.length == 0">
                                    <td colspan="5">
                                        <div class="text-center">
                                            <h3>No approved {{ vm.filterDealType == 'upsell' ? 'upsell' : 'deal' }}s</h3>
                                        </div>
                                    </td>
                                </tr>
                                <tr ng-show="vm.isLoading">
                                    <td colspan="5">
                                        <ng-include src="'/app/common/loading.html'"></ng-include>
                                    </td>
                                </tr>
                            </tbody>
                        </table>
                    </div>
                </div>
            </div>
        </div>
    </div>
</div>
